unic/ucd/name/tests/basic_tests.rs

```diff
@@ -13,12 +13,170 @@ extern crate unic_ucd_name;
 use unic_ucd_name::Name;
 
 #[test]
-fn basic_tests() {
+fn test_name_str() {
+    // == NR1: Hangul Syllable ==
+
+    assert_eq!(
+        Name::of('곲').unwrap().to_string(),
+        "HANGUL SYLLABLE ACF2" // FIXME: use decomposed jamo short names instead
+    );
+
+    // == NR2: Ideographs ==
+
+    assert_eq!(
+        Name::of('言').unwrap().to_string(),
+        "CJK UNIFIED IDEOGRAPH-8A00"
+    );
+    assert_eq!(
+        Name::of('\u{17005}').unwrap().to_string(),
+        "TANGUT IDEOGRAPH-17005"
+    );
+    assert_eq!(
+        Name::of('\u{1B175}').unwrap().to_string(),
+        "NUSHU CHARACTER-1B175"
+    );
     assert_eq!(
-        Name::of('A').expect("No name for A").to_string(),
-        "LATIN CAPITAL LETTER A"
+        Name::of('理').unwrap().to_string(),
+        "CJK COMPATIBILITY IDEOGRAPH-F9E4"
     );
+
+    // == NR3: Explicitly Listed Names ==
+
+    assert_eq!(Name::of('A').unwrap().to_string(), "LATIN CAPITAL LETTER A");
+
+    // == NR4: Null Name ==
+
     assert_eq!(Name::of('\u{10FFFF}'), None);
+}
+
+#[test]
+fn test_name_cmp() {
+    // == NR1 ==
+
+    // FIXME: use decomposed jamo short names instead
+    assert!(Name::of('곲') > Name::of('가'));
+    assert!(Name::of('곲').unwrap().to_string() > Name::of('가').unwrap().to_string());
```

Comment:
Feel like writing a little macro here for the test logic, and only call the macro on test values?

Here are two examples of doing that in our tests:

https://github.com/behnam/rust-unic/blob/c2ed9f869964192f96ac137fc87847aa7cbb1ca5/unic/segment/tests/words_conformance_tests.rs#L28-L39

https://github.com/behnam/rust-unic/blob/c2ed9f869964192f96ac137fc87847aa7cbb1ca5/unic/ucd/common/tests/consistency_tests.rs#L20-L39

Reply:
Fixed :)